### PR TITLE
fix: handle invalid config JSON

### PR DESCRIPTION
**Pull request by nsalamad**
**Note**: put on hold for now

### CODE REVIEW

**Comment by nsalamad** [comment]
Current dependencies on/for this PR:
* main
  * **PR #70** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/70" target="_blank">Graphite</a>  👈
    * **PR #71** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/71" target="_blank">Graphite</a> 
      * **PR #78** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/78" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/dotnet-server-sdk/70?utm_source=stack-comment).

**Review by emmawillis — APPROVED** [review]

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 1]
```diff
@@ -205,10 +204,9 @@ private async Task FetchConfigAsyncWithTask()
 
                 logger.LogError(finalError.ErrorResponse.Message);
                 dvcEventArgs.Errors.Add(finalError);
-            } catch (System.ArgumentNullException e) {
-                // This error can be thrown if the config is in an invalid format 
-                // from the method SetConfig
-                logger.LogError(e.Message);
+            } catch (System.Exception e) {
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: is there not a specific error type you can catch here?

Comment (nsalamad) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: I don't think so. The exception is thrown from https://github.com/DevCycleHQ/assemblyscript-json/blob/main/assembly/decoder.ts#L144. And I think cause it's from assemblyscript, it ends up begin a generic `System.Exception`

Comment (nsalamad) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: catching `WasmtimeException` now instead.

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**